cmd/mesh/profile-dump.go

```diff
@@ -173,6 +174,33 @@ func genICPS(inFilename, profile, setOverlayYAML string, force bool, l *logger)
 	return finalYAML, finalICPS, nil
 }
 
+func genOverlayICPS(filename string) (string, *v1alpha2.IstioControlPlaneSpec, error) {
+	if strings.TrimSpace(filename) == "" {
+		return "", nil, nil
+	}
+
+	b, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return "", nil, fmt.Errorf("could not read from file %s: %s", filename, err)
+	}
+	overlayICPS, _, err := unmarshalAndValidateICP(string(b))
+	if err != nil {
+		return "", nil, err
+	}
+
+	t, err := translate.NewTranslator(binversion.OperatorBinaryVersion.MinorVersion)
+	if err != nil {
+		return "", nil, err
+	}
+
+	overlayYAML, err := component.TranslateHelmValues(overlayICPS, t, "")
```

Comment:
i think this may cause more problems than help. I suggest just using the untranslated input as a mask.

Reply:
I am not 100% sure this is the right thing to do. Let me take a look at how the translator works.

Reply:
You are right. I will fix this.